osu.Game/Screens/Play/MasterGameplayClockContainer.cs

```diff
@@ -86,6 +97,8 @@ private double findEarliestStartTime()
 
         protected override void StopGameplayClock()
         {
+            actualStopTime = GameplayClock.CurrentTime;
```

Comment:
Why does this work? I've read the xmldoc and I can't understand how this being here and _not_ inside the `OnComplete()` callback below translates to the clock seeking to the correct time at unpause? Logically I would think it think it would seek to just under 200ms or so prior to the point where gameplay is fully stopped.

Reply:
In `FrameStabilityClock`, as soon as it sees `IsPaused` it stops processing gameplay. So yes, you are correct in thinking that this will seek 200ms prior, but from the gameplay perspective this is where it wants to be.

I intend on addressing this in the refactor I have planned (ie. optimally no backward seek should be required. it should always go forward and everything should see that ongoing time, including `FrameStabilityClock`).

Reply:
Today I learnt that gameplay is paused immediately.

Reply:
Yeah, as I mentioned somewhere recently, it's not how I want it to be, but I believe it was done to avoid entering a `Failed` state after `Paused`. If you remember that debacle.

Reply:
Perhaps renaming `StopGameplayClock` to `PauseGameplayClock` might indicate it a bit further, as I would've also thought `StopGameplayClock` to be called when the clock should actually just be stopped.